src/tfr_dumping/src/dumping_action_server.cpp

```diff
@@ -124,8 +137,22 @@ class Dumper
                     velocity_publisher.publish(cmd);
                 }
             }
-            //TODO raise the bin
-            stopMoving();
+
+            ROS_INFO("dumping action server detected light raising bin");
+            std_msgs::Float64 bin_cmd;
+            bin_cmd.data = BIN_EXTENDED;
+            tfr_msgs::QuerySrv query;
+            ros::Rate rate(10);
+            while (!server.isPreemptRequested() && ros::ok())
+            {
+                ros::service::call("is_bin_extended", query);
+                if (query.response.data)
+                    break;
+                bin_publisher.publish(bin_cmd);
+                stopMoving();
```

Comment:
We can move stop moving before the loop. I was just paranoid about any extra messages lingering in the network, and messing up our navigational aid. As far as repeatedly sending stop messages, I could take it or leave it. We are not in performance critical code, but it does look a little sloppy and excessive in the loop. 

As far as raising the bin. We use a position controller, so we make a message with the angle in radians that we want the bin to move to, publish it to the controller's `command` topic and the controller will work with the control layer to take care of the rest.

Most controllers require messages to be resent periodically or else they will consider their command "stale". Once it reaches it's goal, it will just hold position.

Let me know if that clarified what you wanted answered about the control loop.

Reply:
For some reason I didn't see the `bin_publisher.publish()` call.